EEParser.py

```diff
@@ -0,0 +1,39 @@
+import os
+import re
+import SelfTexting
+from time import sleep
+
+class WarframeLogParser:
+    def __init__(self, log_path):
+        self.log_path = log_path
+
+    def clean_username(self, username):
+        # Encode and decode to clean up some garbage unicode that gets attached to the end of the username
+        return username.encode("ascii", "ignore").decode("ascii")
+
+    def process_line(self, line):
+        # Adding tab with channel name: FAlextremeYT to index 10
+        # This line fits all our criteria, it gets called when you receive a new whisper from someone,
+        # the username is in cleartext, and any tabs with an index of 6 or above are reserved for usernames so no false alarms from other chat windows.
+        result = re.search(r'(F)(.*\S)( to index (?:[6-9]|[1-9]\d)$)', line)

```

Comment:
So, not in all cases will direct messages appears in index >= 6.

With the default Warframe social settings this will be true, but if you have opted to hide any of the chats then the index will be different 

Options -> Social -> Chat
![image](https://github.com/akmayer/Warframe-Algo-Trader/assets/23193271/7577808e-e586-48c2-b615-868687a02af2)

If you disable all the available chats to disable, you're left with squad and clan chat. Index 1 and 2, so if someone were to DM you, it would result in being added as index 3.

eg. `30859.516 Script [Info]: ChatRedux.lua: ChatRedux::AddTab: Adding tab with channel name: FFmusci to index 3`

![image](https://github.com/akmayer/Warframe-Algo-Trader/assets/23193271/14227170-850c-4598-8700-a4fd3bf54506)

As for a clean solution, i'm not too sure, the minimum will always be index >= 3, due to not being able to turn off squad and clan. The rest of the predefined chats, do have a structure

```
31416.352 Script [Info]: ChatRedux.lua: ChatRedux::AddTab: Adding tab with channel name: G_EN_EU to index 3
31416.354 Script [Info]: ChatRedux.lua: ChatRedux::AddTab: Adding tab with channel name: R_EN_EU to index 4
31416.354 Script [Info]: ChatRedux.lua: ChatRedux::AddTab: Adding tab with channel name: T_EN_EU_WIN to index 5
31416.354 Script [Info]: ChatRedux.lua: ChatRedux::AddTab: Adding tab with channel name: Q_EN_EU to index 4
```

could have a regex of some sort to attempt in detecting these chats, accounting for different regions and platforms, something like `R_[A-Z]{2}_[A-Z]{2}` would detect region chat. But then who's to say that `R_EN_EN` isn't someone's warframe name xD, this would fail in that scenario.

The alternate is to document this issue and ensure people have all chats enabled.

Reply:
Correct me if I'm wrong but it seems like region chat is prepended with the G_, the questions and answers chat are prepended with Q_, recruiting chat is prepended with R_, trade chat is prepended with T_, and whispers seem to be prepended with F. I only tested this briefly but if it's consistent, even if someone's name is `R_EN_EN`, it would show up in the EELog as a tab with `FR_EN_EN`